dashboard/app/helpers/levels_helper.rb

```diff
@@ -41,8 +41,16 @@ def set_videos_and_blocks_and_callouts_and_instructions
     select_and_track_autoplay_video
 
     if @level.is_a? Blockly
-      @toolbox_blocks = @toolbox_blocks || @level.toolbox_blocks
-      @start_blocks = initial_blocks(current_user, @level) || @start_blocks || @level.start_blocks
+      @toolbox_blocks =
+        @toolbox_blocks ||
+        @level.try(:project_template_level).try(:toolbox_blocks) ||
+        @level.toolbox_blocks
+
+      @start_blocks =
+        initial_blocks(current_user, @level) || # check if this level inherits solution from previous level
+        @start_blocks ||
+        @level.try(:project_template_level).try(:start_blocks) ||
+        @level.start_blocks
```

Comment:
hm, I deliberately did it this way to visually compare/contrast toolbox_blocks and start_blocks by formatting the last 3 statements in the same way so it's obvious that it's the same logic except or the initial_blocks line. pulling up the first lines of the rvalue would lose that. (same reason I didn't do toolbox_blocks as a ||= which would be shorter)

If you still prefer same line I can do that of course